Doc/reference/executionmodel.rst

```diff
@@ -133,14 +133,15 @@ determined by scanning the entire text of the block for name binding operations.
 See :ref:`the FAQ entry on UnboundLocalError <faq-unboundlocalerror>`
 for examples.
 
-If the :keyword:`global` statement occurs within a block, all uses of the names
+The :keyword:`global` statement must precede all uses of the listed names.
+If the global statement occurs within a block, all uses of the names
 specified in the statement refer to the bindings of those names in the top-level
 namespace.  Names are resolved in the top-level namespace by searching the
-global namespace, i.e. the namespace of the module containing the code block,
+global namespace, the namespace of the module containing the code block,
```

Comment:
Removing "i.e." makes this wrong: the global namespace is the same as the module namespace here.

Reply:
That's what I am implying here. The next part of the sentence is written in the same way.

"the builtins namespace, the namespace of the module builtins" implying the builtins namespace and the namespace of the module builtins are the same.

Reply:
I see; I don't think that is clear on reading the sentence.

Reply:
How about this version of the sentence?

Names are resolved in the top-level namespace by searching the global namespace which is the namespace of the module containing the code block, and the builtins namespace which is the namespace of the module builtins.

Reply:
That works though I'd put commas before "which". I think the existing wording is fine though, so I'd prefer to keep it.

Reply:
Reverted it.

But the fact that we both needed secondary clarification is a sign this sentence could be written better.